src/source/PeerConnection/Rtp.c

```diff
@@ -129,24 +131,64 @@ STATUS transceiverOnBandwidthEstimation(PRtcRtpTransceiver pRtcRtpTransceiver, U
     return retStatus;
 }
 
+STATUS updateEncoderStats(PRtcRtpTransceiver pRtcRtpTransceiver, PRtcEncoderStats encoderStats)
```

Comment:
Maybe I am missing something. Why do we need encoderStats? Is it only for the application to update it? 
Can't we let them add these details to `outboundRtpStreamStats.frameHeight` directly? It will avoid the usage of encoderStats and outboundRtpStreamStats object is anyways something customer will have to pass while calling rtcPeerConnectionGetMetrics. We cans imply add a note saying the information needs to be updated by the customer.

Reply:
i would prefer to not let user modify stats directly because of locking

Reply:
The main issue is that our solution does not control the encoder like some WebRTC implementations that wrap the encoder itself. In reality, the encoder path and the stats path might be entirely different entities so having a separate public API is a good choice. This said, I am not sure how prolific is the encoder stats out in the wild and how many customers will actually supply these stats.

Reply:
i will supply the stats for sure

Reply:
if the user doesnt supply stats, nothing bad happens, the fields affected by the stats wont be populated so the user may populate them later after calling appropriate metrics api method. 
example client who doesnt want to update metrics:
```
RtcOutboundRtpStreamStats give_me_stats_here;
getRtpOutboundStats(..., &give_me_stats_here);
give_me_stats_here.totalEncodeTime = client_knows_where_to_get_it_from(); //we return zero in the field
...
use_stats(give_me_stats_here)
```